File: 278_first_bad_version_number.cpp
// The API isBadVersion is defined for you.
// bool isBadVersion(int version);

class Solution {
public:
    int firstBadVersion(int n) {
    
     return badVersion(1,n);
    }
   int badVersion( int s, int e){
       if(s==e){
           return s;
       }
       
       
       int mid=s+(e-s)/2;
       
       if(isBadVersion(mid)){
          return badVersion(s,mid);
       }
       else{
          return badVersion(mid+1,e);
       }
       
   }
  
};

File: 278_first_bad_version_number.py
# The isBadVersion API is already defined for you.
# @param version, an integer
# @return an integer
# def isBadVersion(version):

class Solution:
    def firstBadVersion(self, n):
        s=1
        e=n
        while(s<n):
            mid=int(s+(e-s)/2)
            if(isBadVersion(mid)):
                e=mid
            
            else:
                s=mid+1
                
        
        return s
    
    
        
